Model/RatepayInvoicePaymentMethod.php

```diff
@@ -461,4 +473,133 @@ private function _areAddressesAllowed($sBillingCountryId, $sShippingCountryId)
             $sBillingCountryId === $sShippingCountryId
         );
     }
+
+    /**
+     *
+     * @inheritdoc
+     *
+     * @param string                           $sAction
+     * @param \Wirecard\Oxid\Model\Transaction $oParentTransaction
+     * @param array|null                       $aOrderItems
+     *
+     * @return Transaction
+     *
+     * @since 1.2.0
+     */
+    public function getPostProcessingTransaction($sAction, $oParentTransaction, $aOrderItems = null)
+    {
+        $oTransaction = new RatepayInvoiceTransaction();
+        $aBasketResult = $this->_createPostProcessingBasket($oParentTransaction, $aOrderItems);
+
+        /**
+         * @var $oBasket Basket
+         */
+        $oBasket = $aBasketResult[0];
+        $oTransaction->setBasket($oBasket);
+        $oTransaction->setAmount(new Amount(
+            $aBasketResult[1],
+            $oBasket->getTotalAmount()->getCurrency()
+        ));
+        return $oTransaction;
+    }
+
+    /**
+     * @param \Wirecard\Oxid\Model\Transaction $oParentTransaction
+     * @param array|null                       $aOrderItems
+     *
+     * @return array
+     *
+     * @since 1.2.0
+     */
+    private function _createPostProcessingBasket($oParentTransaction, $aOrderItems)
+    {
+        $oBasket = new Basket();
+
+        $oXmlBasket = simplexml_load_string($oParentTransaction->getResponseXML());
```

Comment:
brake the method into up into more methods like loadBasketFromXml, then for the creation of the basket you have a basket helper, feed him with the data and reuse the code that is there. Im also confused that the _getItemsToAddToBasket() is getting a basket entity and an xml basket, its the same data send in two formats? I suggest the refactoring of it.

Reply:
the 2 baskets are needed because of the issue with the paymentSdk and the tax rate. we can get rid of that handling once the paymentSdk is fixed.

Reply:
got rid of the new xml basket with the new paymentsdk version